frontend/src/utils/QuestionUtils.ts

```diff
@@ -32,3 +29,20 @@ export const shouldReorder = (
   }
   return false;
 };
+
+export const updatedQuestionElement = (
+  id: string,
+  updatedData: string,
+  error: string,
+  prevElements: QuestionElement[],
+) => {
+  const indexToUpdate = prevElements.findIndex((element) => element.id === id);
+  return update(prevElements, {
+    [indexToUpdate]: {
+      $merge: {
+        data: updatedData,
+        error: updatedData.length > 800 ? error : "",
```

Comment:
hmm maybe we move the error logic into the components themselves? since this length check only applies to text and question text, but if we want to use for other components will be different

Reply:
oops I’m silly, ty!